[17.0] [MIG] partner_industry_secondary

mmrondon opened the pull request.

MiquelRForgeFlow committed:
partner_sector -> partner_industry_secondary

MiquelRForgeFlow committed:
Migration scripts (from v10 to v11)

Commit:
partner_industry_secondary: Migration to v12

badbole committed:
Translated using Weblate (Croatian)

Currently translated at 83.3% (15 of 18 strings)

Translation: partner-contact-12.0/partner-contact-12.0-partner_industry_secondary
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-12-0/partner-contact-12-0-partner_industry_secondary/hr/

CarlesAntoli committed:
Translated using Weblate (Spanish)

Currently translated at 88.9% (16 of 18 strings)

Translation: partner-contact-12.0/partner-contact-12.0-partner_industry_secondary
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-12-0/partner-contact-12-0-partner_industry_secondary/es/

sergio-teruel committed:
partner_industry_Secondary: Make api constrains multi to avoid error when create a company with 2 contacts

skeller1 committed:
[IMP] partner_industry_secondary: black, isort, prettier

skeller1 committed:
[MIG] partner_industry_secondary: Migration to 13.0

SurachetS committed:
[IMP] : black, isort, prettier

SurachetS committed:
[MIG] partner_industry_secondary: Migration to 14.0

SilvioC2C committed:
[IMP] partner_industry_secondary: black, isort, prettier

SilvioC2C committed:
[MIG] partner_industry_secondary: Migration to 15.0

pablontura committed:
Translated using Weblate (Catalan)

Currently translated at 100.0% (17 of 17 strings)

Translation: partner-contact-15.0/partner-contact-15.0-partner_industry_secondary
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-15-0/partner-contact-15-0-partner_industry_secondary/ca/

victor-champonnois committed:
[MIG] partner_industry_secondary: Migration to 16.0

victor-champonnois committed:
fixup! [MIG] partner_industry_secondary: Migration to 16.0

Commit:
[UPD] Update partner_industry_secondary.pot

OCA-git-bot committed:
[UPD] README.rst

victor-champonnois committed:
Translated using Weblate (French)

Currently translated at 100.0% (17 of 17 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_industry_secondary
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_industry_secondary/fr/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (17 of 17 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_industry_secondary
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_industry_secondary/es/

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: partner-contact-16.0/partner-contact-16.0-partner_industry_secondary
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_industry_secondary/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (17 of 17 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_industry_secondary
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_industry_secondary/es/

mymage committed:
Translated using Weblate (Italian)

Currently translated at 100.0% (17 of 17 strings)

Translation: partner-contact-16.0/partner-contact-16.0-partner_industry_secondary
Translate-URL: https://translation.odoo-community.org/projects/partner-contact-16-0/partner-contact-16-0-partner_industry_secondary/it/

marielejeune committed:
[IMP] partner_industry_secondary: add unaccent=False on parent_path field
As asked by Odoo in a warning in _check_parent_path method
https://github.com/odoo/odoo/blob/16.0/odoo/models.py#L2615

OCA-git-bot committed:
[BOT] post-merge updates

mmrondon committed:
[IMP] partner_industry_secondary: pre-commit auto fixes

DucTruongKomit commented on partner_industry_secondary/views/res_partner_view.xml:
```diff
@@ -0,0 +1,79 @@
+                <field
+                    name="secondary_industry_ids"
+                    widget="many2many_tags"
+                    invisible="is_company == False and show_partner_industry_for_person == False"
```
I imagine that the user should set this field after they set the field industry_id 1st. Because the this field got domain [('id', '!=', industry_id)]

So, we should hide this field also if industry_id = False

For example:
I set Partner A with industry_id then, display this field, so the user may think about their options on secondary_industry_ids

Do you agree with me?

mmrondon replied:
hello @DucTruongKomit 
Yes, your comment makes sense, but in this case the module already has a restriction in case the same industry is selected as main and secondary, only possible if a secondary industry is selected first.

![validationError](https://github.com/OCA/partner-contact/assets/89471929/1f7e808e-f270-4f7c-8b16-d7379ff94573)

DucTruongKomit reviewed (commented):
Hello! Do you agree with me on my comments?

DucTruongKomit reviewed (approved):
LGTM! All gud!

HaraldPanten commented:
@mmrondon You have too many commits for this migration. Don't you?

Could you clean the commits and leave only what we need for the migration?

THX.

mmrondon committed:
[MIG] partner_industry_secondary: Migration to 17.0

[IMP] Fix pre-commit and unit tests

[IMP] partner_industry_secondary: black, isort, prettier

[IMP] partner_industry_secondary: Update unit tests

[IMP] partner_industry_secondary: black, isort, prettier

Tisho99 commented on partner_industry_secondary/views/res_config_settings.xml:
```diff
@@ -0,0 +1,49 @@
+        <field name="inherit_id" ref="base.res_config_settings_view_form" />
+        <field name="arch" type="xml">
+            <div id="contacts_settings" position='inside'>
+                <div class="row mt16 o_settings_container">
+                    <div class="col-lg-6 o_setting_box">
+                        <div class="o_setting_left_pane">
+                            <field name="group_use_partner_industry_for_person" />
+                        </div>
+                        <div
+                            class="o_setting_right_pane"
+                            id="partner_industry_secondary_show_form_settings"
+                        >
+                            <label
+                                string="Enable industries for individuals"
+                                for="group_use_partner_industry_for_person"
+                            />
+                            <div class="text-muted">
+                                Enable industries for individuals
+                             </div>
+                            <div class="content-group">
+                                <div
+                                    class="mt16"
+                                    invisible="group_use_partner_industry_for_person == False"
+                                >
+                                    <button
+                                        name="%(base.res_partner_industry_action)d"
+                                        icon="fa-arrow-right"
+                                        type="action"
+                                        string="Industries"
+                                        class="btn-link"
+                                    />
+                                </div>
+                            </div>
+                        </div>
+                    </div>
+                </div>
```
```suggestion
                <block>
                    <setting string="Enable industries for individuals" help="Enable industries for individuals">
                        <field name="group_use_partner_industry_for_person" />
                    </setting>
                </block>
```

Tisho99 commented on partner_industry_secondary/views/res_partner_view.xml:
```diff
@@ -0,0 +1,81 @@
+                    name="industry_id"
+                    invisible="is_company == False and show_partner_industry_for_person == False"
+                />
+                <field name="show_partner_industry_for_person" invisible="1" />
```
```suggestion
                <field name="show_partner_industry_for_person" column_invisible="1" />
```

Tisho99 reviewed (changes requested):
Hi @mmrondon 

I have made a functional and code review and this migration looks good.

However, i found that you haven't applied some steps of the [migration guide](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0)

Can you review and apply my suggestions in your last commit?

Then this will be ready for merge

Thank you!

HaraldPanten commented:
Superseded by https://github.com/OCA/partner-contact/pull/1761
